extensions/emoji/src/emoji.tsx

```diff
@@ -95,14 +97,7 @@ export default function Main(): ReactElement {
       }
     >
       {!isLoading
-        ? filterList(
-            [
-              { category: "Recently Used", emojis: recentlyUsed },
-              { category: category || "Emojis", emojis: list },
-            ],
```

Comment:
Rather than doing all these changes wouldn't it be simpler to add this section conditionally?

```
            [
              !searchText && { category: "Recently Used", emojis: recentlyUsed },
              { category: category || "Emojis", emojis: list },
            ].filter(Boolean)

Reply:
To do it like this we'd have to cast it like so

```typescript
[
  !searchText && { category: "Recently Used", emojis: recentlyUsed },
  { category: category || "Emojis", emojis: list },
].filter(Boolean) as Category[]
```

or do

```typescript
[
  ...(!searchText ? [{ category: "Recently Used", emojis: recentlyUsed }] : []),
  { category: category || "Emojis", emojis: list },
].
```

I was afraid of putting too much logic in this method arguments, and thought it might be better off filter method. I'm of course open to either of these approaches though, which do you prefer?

Reply:
I'd like not to overload the filter function with non-filter related logic. I think your first example would make sense, it would also reduce a lot the line changes of this PR.

Reply:
Alternatively we could have an additional function that is responsible of adding the recently used category.

```
addRecentlyUsed([{ category: category || "Emojis", emojis: list }], !searchText).map(... etc
```